popup/searchResults.js

```diff
@@ -0,0 +1,251 @@
+'use strict';
+
+/**
+ * Library for interacting with userstyles.org
+ * @returns {Object} Includes fetch() method which promises userstyles.org resources.
+ */
+const UserStylesAPI = (() => {
+  return {fetch}
+
+  /**
+   * Fetches (and JSON-parses) the result from a userstyles.org API
+   * @param {string} path Path on userstyles.org (e.g. "/api/v1/styles/search")
+   * @param {string} queryParams Query parameters to send in search request (e.g. "key=value&name=that)".
+   * @return {Object} Response object from userstyles.org
+   */
+  function fetch(path, queryParams) {
```

Comment:
We already have `download()` and `tryJSONparse()`

Reply:
Thanks for telling me about ESLint. It found a number of warnings/errors that I have since fixed. The ESLint fixes will show up in my next PR update.

---------

I think I can `download()`  the `/styles/chrome/*.json` file. And I'll use `tryJSONparse()` when possible.

However, `download()` [forces the method to `POST` if the URL has a `search` field](https://github.com/openstyles/stylus/blob/master/js/messaging.js#L387). The UserStyles search API (`/api/v1/styles/search?search=...&page=...`) requires URL Search Parameters, and the request has to be a `GET`.

Performing a `POST` on the search API returns a `405` error:

```bash
$ curl -X POST "https://userstyles.org/api/v1/styles/search?search=userstyles.org&page=1&per_page=2&category=userstyles.org" -vs 2>&1 | grep "< HTTP"
< HTTP/1.1 405 Method Not Allowed
```

I see [`content/install-user-css.js` has it's own `fetchText()` method](https://github.com/openstyles/stylus/blob/master/content/install-user-css.js#L7-L16). But it's encapsulated within `createSourceLoader`.

Should I:

1. Keep `UserStylesAPI.fetch()` in `search-results.js` since it can't use `download()`? Or,
2. Extract `fetchText()` into some utility class, to be used by both `install-user-style.js` and `search-results.js`?

I'm leaning towards Option 1 since the userstyles.org API may require specific headers in the future (e.g. `Content-Type`, `Accept`, `POST` instead of `GET`), and `fetchText()` doesn't currently support customizing headers.

Reply:
leave it as is, we'll deal with this later

Reply:
Bhai WiFi password hack show video give me